lib/spack/spack/patch.py

```diff
@@ -36,10 +36,11 @@ def apply_patch(stage, patch_path, level=1, working_dir="."):
     """
     git_utils_path = os.environ.get("PATH", "")
     if sys.platform == "win32":
-        git = which_string("git", required=True)
-        git_root = git.split("\\")[:-2]
-        git_root.extend(["usr", "bin"])
-        git_utils_path = os.sep.join(git_root)
+        git = which_string("git")
+        if git:
+            git_root = git.split(os.sep)[:-2]
```

Comment:
Can we use `pathlib.Path.parent.parent`?  I want to avoid possible confusion about the behavior of splitting strings like `C:\\` (I think that technically works here, but `pathlib.Path` would be more clear).

Reply:
We should point to this if we ever have to justify the use of pathlib in the codebase, it made the code orders of magnitude cleaner and easier to understand at a glance.

Reply:
Amended and pushed.